test/automate/automate-utils.js

```diff
@@ -14,285 +14,292 @@
  * - Click on the "Tests" button to see the results.
  * - Grab a screenshot of the results and save it to disk.
  *
- * Note that we tried to get headless Chrome to work running these tests but
- * there were intermitant failures, so that is not a viable option as of July
- * 2017. Perhaps a later version will work more reliably. In which case, we
- * no longer need Xvfb in the travis.yml file.
  */
 
+// Used to save screenshots.
+const fs = require('fs');
+const path = require('path');
+
 let screenShotUniq = 0;
 
- // Grab a screenshot and write to disk.
- // TODO: Do we want to grab screenshots for success?
- const saveScreenshot = function(driver, screenshotDir, name, type) {
-
-   const fileName =
-     `test-result-${screenShotUniq}-${name.replace(/ /g, '-')}-${type}.png`;
-
-   screenShotUniq++;
-
-   return driver.takeScreenshot()
-   .then(function(data) {
-     if (data) {
-       var base64Data = data.replace(/^data:image\/png;base64,/, '');
-       fs.writeFile(
-         path.join(screenshotDir, fileName),
-         base64Data,
-         'base64',
-         function(err) {
-          if (err) { console.log(err); }
-         }
-       );
-     }
-   });
- };
-
-// Chromedriver is used by Selenium and requiring the module avoids having to
-// download and install the executable from Google along with setting the path.
-/* eslint no-unused-vars: ["error", { "varsIgnorePattern": "chromedriver" }] */
+// Chromedriver and geckodriver are used by Selenium and requiring the module
+// avoids having to download and install the executable from Google
+// along with setting the path.
+/* eslint no-unused-vars: ["error", {
+    "varsIgnorePattern": "chromedriver|geckodriver"
+   }] */
 let chromedriver = require('chromedriver');
+let geckodriver = require('geckodriver');
 
 let webdriver = require('selenium-webdriver'),
     By = webdriver.By,
     until = webdriver.until;
 
-// Used to save screenshots.
-const fs = require('fs');
-const path = require('path');
-
-exports.doesProjectPassTests = function(name, URL) {
+exports.doesProjectPassTests = (browser, name, URL) => {
 
   // If elements don't appear within 1 minute, it's usually an error.
   const elementTimeout = 60000;
 
   // If all of the project tests pass, this is set to true.
   let success = false;
 
-  const chrome = require('selenium-webdriver/chrome');
+  // headless mode for chrome and firefox is enabled by default
+  let headless = true;
+  if (typeof process.env.HEADLESS !== 'undefined') {
+    // eslint-disable-next-line eqeqeq
+    headless = process.env.HEADLESS == true;
```

Comment:
Can you please use triple equal (===) here? I'm suprised ESLint didn't warn you about this.